request.go

```diff
@@ -22,8 +22,8 @@ const (
 type BotClient interface {
 	// RequestWithContext submits a POST HTTP request a bot API instance.
 	RequestWithContext(ctx context.Context, token string, method string, params map[string]string, data map[string]FileReader, opts *RequestOpts) (json.RawMessage, error)
-	// TimeoutContext calculates the required timeout contect required given the passed RequestOpts, and any default opts defined by the BotClient.
-	TimeoutContext(opts *RequestOpts) (context.Context, context.CancelFunc)
+	// TimeoutContext wraps the existing context with the timeout from the provided RequestOpts, or any default opts defined by the BotClient.
+	TimeoutContext(ctx context.Context, opts *RequestOpts) (context.Context, context.CancelFunc)
```

Comment:
BTW, just curious about this method 😄 

What's your consideration to define this method separately?
I think we can pass timeout parameter to https://github.com/PaulSonOfLars/gotgbot/blob/v2.0.0-rc.29/bot.go#L42 to achieve the same effect and even if we need ctx to do other things we can do it in `RequestWithContext`.

Reply:
Well, two reasons: 
1) bot-wide timeouts
2) backwards compatibility 

I do agree it would be nice if it wasn't part of it at this point! Might look into it to see if it still needs to be part of the interface though...

Reply:
Got. thanks for replying

Reply:
Managed to clean it up - you were right, doesn't need to be there after all :)